### PR TITLE
show notifications on background backup errors

### PR DESCRIPTION
**Pull request by fyfrey**
- shows a notification if the background backup failed (e.g. connecting to server failed) 
- shows only 1 notification regardless how often this failure occurred (automatic retries are performed in exponentially increasing intervals)
- shows individual error notifications if specific files could not be uploaded (because the API said so)
- automatically clears all errors once the background backup completed successfully

How to test this:

1. Have background backup enabled & configure immediate notifications in app settings
2. Shutdown the server (`docker compose stop immich-proxy` is enough)
3. Take a new picture
4. Wait until background backup runs; it should produce a notification
5. Restart server
6. Wait until the service retries the upload (60 seconds)
7. Check that error notification is gone and pictures uploaded to the server

### CODE REVIEW

**Comment by bo0tzz** [comment]
Is it possible to add a retry button on the notification that shortcuts the backoff timer? Useful when you temporarily lost connection but know you can run again.

**Comment by zoodyy** [comment]
> Is it possible to add a retry button on the notification that shortcuts the backoff timer? Useful when you temporarily lost connection but know you can run again.

Did some research on that. Seems it is a) quite a lot of work and b) not possible not force a retry other than stop&start which has its own issues. Thus, I do not plan to add such a retry button in the near future. However, if some more experienced Android dev knows how to do it, I'd like to have the button as well :smile:

File: mobile/lib/modules/backup/services/backup.service.dart [hunk 1]
```diff
@@ -147,7 +134,8 @@ class BackupService {
     return result;
   }
 
-  Future<List<AssetEntity>> _removeAlreadyUploadedAssets(
+  /// Returns a new list of assets not yet uploaded
```
Comment (alextran1502) [inline mobile/lib/modules/backup/services/backup.service.dart:137]: Let's ask some comments for this method that it is currently being used only for background service.

**Comment by alextran1502** [comment]
I realized that the API to check for duplicate assets on my part was a bad design. We should send in all the asset's IDs that need to be checked and return the Ids that are duplicated rather than we have to make a single request for each asset. 

I will probably make a PR to address this issue sometime soon.

**Comment by zoodyy** [comment]
> We should send in all the asset's IDs that need to be checked and return the Ids that are duplicated rather than we have to make a single request for each asset.

Absolutely! My current approach is just a temporary workaround until either of us changes the API in another PR

**Comment by alextran1502** [comment]
Look like we are good to go. Let's resolve the conflict then I will merge it in

**Pull request merged**